webapp/packages/core-blocks/src/FormControls/useInputAutocomplete.ts

```diff
@@ -0,0 +1,154 @@
+  filter?: (suggestion: InputAutocompleteProposal, lastWord?: string) => boolean;
+}
+
+export interface InputAutocompleteProposal {
+  displayString: string;
+  replacementString: string;
+  icon?: string;
+  title?: string;
+  score?: number;
+}
+
+const INPUT_DELAY = 300;
+
+export const useInputAutocomplete = (
+  inputRef: RefObject<HTMLInputElement | HTMLTextAreaElement>,
+  { sourceHints, matchStrategy = 'startsWith', filter }: InputAutocompleteOptions,
+) => {
+  const state = useObservableRef(
+    () => ({
+      input: inputRef.current?.value as string | undefined,
+      selectionStart: inputRef.current?.selectionStart ?? null,
+      selectionEnd: inputRef.current?.value?.length ?? null,
+      replaceCurrentWord(replacement: string) {
+        const input = this.inputRef.current;
+
+        if (!input) {
+          return;
+        }
+
+        const cursorPosition = this.selectionStart;
+        const words = this.input?.split(' ');
+
+        if (!isNotNullDefined(words) || !isNotNullDefined(cursorPosition) || !isNotNullDefined(this.currentWord)) {
+          return;
+        }
+
+        const start = cursorPosition - this.currentWord.length;
+        const end = cursorPosition;
+
+        this.input = this.input?.slice(0, start) + replacement + this.input?.slice(end);
+        this.selectionStart = start + replacement.length;
+        this.selectionEnd = start + replacement.length;
+
+        input.value = this.input;
+        input.focus();
+      },
+      get currentWord() {
+        const cursorPosition = this.selectionStart;
+
+        if (!cursorPosition) {
+          return '';
+        }
+
+        const substring = this.input?.slice(0, cursorPosition);
+
+        if (!substring) {
+          return '';
+        }
+
+        return (
+          substring
+            .split(' ')
+            .at(-1)
+            ?.replace(/[^\w\s]|_/g, '') ?? ''
+        );
+      },
+      get filteredSuggestions() {
+        if (!this.currentWord) {
+          return [];
+        }
+
+        return this.sourceHints
+          .filter(suggestion => {
+            const values = [suggestion.displayString.toLocaleLowerCase(), suggestion.replacementString.toLocaleLowerCase()];
+            const isEqual = values.some(value => value === this.currentWord?.toLocaleLowerCase());
+
+            if (!this.currentWord || isEqual) {
+              return false;
+            }
+
+            return (
+              (this.matchStrategy === 'startsWith' &&
+                values.some(value => isNotNullDefined(this.currentWord) && value.startsWith(this.currentWord))) ||
+              (this.matchStrategy === 'contains' && values.some(value => isNotNullDefined(this.currentWord) && value.includes(this.currentWord))) ||
+              (this.matchStrategy === 'fuzzy' &&
+                values.some(value => isNotNullDefined(this.currentWord) && isFuzzySearchable(this.currentWord, value)))
+            );
+          })
```

Comment:
```suggestion
const isMatch = () => {
  const { matchStrategy, currentWord } = this;

  if (!isNotNullDefined(currentWord)) {
    return false;
  }

  const matchFunctions: { [key: InputAutocompleteStrategy]: (value: string) => boolean } = {
    startsWith: value => value.startsWith(currentWord),
    contains: value => value.includes(currentWord),
    fuzzy: value => isFuzzySearchable(currentWord, value),
  };

  return values.some(matchFunctions[matchStrategy]);
};
```


Maybe like that? And we checked thus.currentWord just above, so maybe we can skip this check:
```
 if (!isNotNullDefined(currentWord)) {
    return false;
  }
```

Reply:
done